rust_src/src/buffers.rs

```diff
@@ -125,34 +125,36 @@ impl LispBufferRef {
     }
 
     #[inline]
-    pub fn fetch_byte(&self, n: ptrdiff_t) -> u8 {
+    pub fn fetch_byte(self, n: ptrdiff_t) -> u8 {
         let offset = if n >= self.gpt_byte() {
             self.gap_size()
         } else {
             0
         };
 
-        unsafe { *(self.beg_addr().offset(offset + n - self.beg_byte())) as u8 }
+        let mut clone = self;
+        unsafe { *(clone.beg_addr().offset(offset + n - self.beg_byte())) as u8 }
     }
 
     #[inline]
-    pub fn fetch_multibyte_char(&self, n: ptrdiff_t) -> c_int {
+    pub fn fetch_multibyte_char(self, n: ptrdiff_t) -> c_int {
         let offset = if n >= self.gpt_byte() && n >= 0 {
             self.gap_size()
         } else {
             0
         };
+        let mut clone = self;
```

Comment:
Use `mut self` in the signature instead. But it'll become unnecessary after the other change.